modal/experimental.py

```diff
@@ -13,3 +31,81 @@ def get_local_input_concurrency():
     """Get the container's local input concurrency. Return 0 if the container is not running."""
 
     return _ContainerIOManager.get_input_concurrency()
+
+
+# START Experimental: Container Networking
+
+
+class _GroupedFunction(typing.Generic[P, ReturnType, OriginalReturnType], _Object, type_prefix="gf"):
+    def __init__(self, f: _Function, size: int):
+        self.f = synchronize_api(f)
+        self.size = size
+
+    def remote(self, *args: P.args, **kwargs: P.kwargs) -> List[ReturnType]:
+        """
+        Calls the function remotely, executing it with the given arguments and returning the execution's result.
+        """
+        worker_handles: List[FunctionCall] = []
+        with modal.Queue.ephemeral() as q:
+            for i in range(self.size):
+                handle = self.f.spawn(*args, **{**kwargs, "rank": i, "size": self.size, "q": q})
+                worker_handles.append(handle)
+        output: list[ReturnType] = []
+        for i, handle in enumerate(worker_handles):
+            output.append(handle.get())
+        return output
+
+    def local(self, *args: P.args, **kwargs: P.kwargs) -> ReturnType:
+        raise NotImplementedError("Grouped function cannot be run locally")
+
+    def spawn(self, *args: P.args, **kwargs: P.kwargs) -> ReturnType:
+        raise NotImplementedError("Grouped function cannot be spawned")
```

Comment:
Do you need to add `NotImplementedError` for `map`, etc.? Unfortunately `Function` has a lot of methods :(

Reply:
Fixed. Some of these we could possibly implement (eg map), but it would require you to build some sort of wrapper so Grouped Function calls are not synchronous